ArduPlane/ArduPlane.cpp

```diff
@@ -963,6 +963,28 @@ bool Plane::flight_option_enabled(FlightOptions flight_option) const
     return g2.flight_options & flight_option;
 }
 
+#if AP_SCRIPTING_ENABLED
+// this implements the virtual function defined in AP_Vehicle for Plane to set the desired airspeed in m/s
+bool Plane::set_desired_airspeed(float airspeed_new)
+{
+    if (control_mode->is_guided_mode() || control_mode == &mode_auto) {
```

Comment:
As it is currently implemented those are the only two modes that this command works in, hence the check. Totally agree that it should work in more modes in the future. A virtual mode method makes sense. But we don't necessarily have to have that before merging this.

Reply:
I agree with @tridge  but for a different reason. Regardless what modes can/can't be overridden, it shouldn't be code in this API layer that prevents something from working that might later change at a lower level. The mode knows if it can be overridden or not, so creating a can_override_airspeed() on the mode is a better way to implement this check.

Reply:
I've changed it to call a "working" version of can_override_airspeed() that returns the correct values based on current functionality.